### PR TITLE
[make init] update submodule's remotes

### PR DESCRIPTION
**Pull request by yxieca**
**- Why I did it**
I am using a long standing checked out repo to build dev images. From time to time the repo is too out-sync'ed with remotes and some submodule stay behind even after make reset and make init call. This causes build to fail unexpectedly.

**- How I did it**
With further tests, I moved the recursive "git remote update' to make reset code block and placed in front of recursive --init. The original change has the ordering wrong but that was hidden by the fact that make init was executed multiple times in master branch and 201911 branch.

The summary is:
- for a newly cloned repo. existing 'make init' is enough.
- for long standing repo, when things are getting too out dated, or when we switching branch, we usually need to do a make reset before build again. Place the fix in make reset makes more sense.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

**- How to verify it**
Build a dev image. Tested it with a newly cloned repo as well.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

### CODE REVIEW

**Comment by yxieca** [comment]
retest broadcom please

File: Makefile.work [hunk 1]
```diff
@@ -283,6 +283,7 @@ showtag:
 
 init :
 	@git submodule update --init --recursive
+	@git submodule foreach --recursive git remote update &>/dev/null
```
Comment (jleveque) [inline Makefile.work:286]: Maybe we should also add this to `make reset` below?

Comment (yxieca) [reply Makefile.work:286]: possibly, I did see make reset didn't cleanup everything once. we could try this on that too.

Comment (yxieca) [reply Makefile.work:286]: Just to clarify @jleveque, are you requesting a change to be added to this PR or suggesting experiment with make reset separately?

**Review by qiluo-msft — DISMISSED** [review]
LGTM. Please also check with other reviewers.

File: Makefile.work (hunk 1 above)
Comment (jleveque) [reply Makefile.work:286]: @yxieca: Either way is fine. If you want to test it and add it you can, otherwise we can take note for a future PR.

Comment (yxieca) [reply Makefile.work]: Thanks Joe, per our discussion offline and my test results. I moved the recursive "git remote update" to make reset and placed in front of --init.

**Review by jleveque — APPROVED** [review]

**Pull request merged**